### PR TITLE
Add first unit test

### PR DESCRIPTION
**Pull request by Salazander**
- use python 3.10 for compatibility reasons
- install spark runtime in dev container
- add unit test to illustrate the inner dev loop

### CODE REVIEW

File: src/common_lib/tests/test_stream_processor.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from pyspark.sql import SparkSession
+from pyspark.sql import types as T
+import pytest
+
+import sys
+sys.path.append('/workspaces/portable-etl/src/')
```
Comment (syedhassaanahmed) [inline src/common_lib/tests/test_stream_processor.py]: I think there is a way to find out absolute path from relative path
```directory = path.path(__file__).abspath()```

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-    "image": "mcr.microsoft.com/devcontainers/python:0-3.11",
+    "image": "mcr.microsoft.com/devcontainers/python:0-3.10",
```
Comment (Salazander) [inline .devcontainer/devcontainer.json]: add comment why 3.11 doesn't work

Comment (Salazander) [reply .devcontainer/devcontainer.json]: https://stackoverflow.com/questions/75048688/picklingerror-could-not-serialize-object-indexerror-tuple-index-out-of-range

File: src/common_lib/tests/test_stream_processor.py (hunk 1 above)
Comment (Salazander) [reply src/common_lib/tests/test_stream_processor.py]: fixed it

**Review by syedhassaanahmed — APPROVED** [review]
🚀

**Pull request merged**